trieste/models/keras/sampler.py

```diff
@@ -173,24 +168,27 @@ def __call__(self, x: TensorType) -> TensorType:  # [N, B, d] -> [N, B, 1]
 
         if self._diversify:
             predicted_means, predicted_vars = self._model.predict(flat_x)  # ([N*B, 1], [N*B, 1])
-            normal = tfp.distributions.Normal(predicted_means, tf.sqrt(predicted_vars))
-            predictions = tf.map_fn(normal.quantile, self._quantiles)  # [B, N*B, 1]
+            predicted_vars = predicted_vars + tf.cast(DEFAULTS.JITTER, predicted_vars.dtype)
+            predictions = predicted_means + tf.sqrt(predicted_vars) * tf.tile(
```

Comment:
does this work? `self._eps` is `[B, L]` so here with tiling you get `[N*B, L]`? but how do you know that the right epsilon is given to the right trajectory?

Wouldn't it be easier and safer to do without flattening? e.g. with `predicted_means` being `[..., B, L]` and eps `[B, L]`, you just need to add dummy dimensions for broadcasting.

Reply:
I verified it manually that it is correct way of tiling it...
we cannot really do it without flatenning, we would need B forward passes with the networks which would be much slower

Reply:
perhaps I can think of a test that would make sure eps are applied correctly...

Reply:
I understand. Yes it would be really important to test this. Possibly you for a test you can assign `[0, 1]` to `eps` and check that one trajectory contains only zeros and the other doesn't contain any?

Reply:
small point: `# [B, N*B, 1]` should be `# [B, N*B, L]` now, correct?

Reply:
I think the existing test mentioned above should be enough, but fair enough, added another one you have suggested `test_ensemble_trajectory_sampler_eps_broadcasted_correctly` - works in eager mode but not with tf.function, probably its hard to set _eps externally in that mode - @uri-granta can you perhaps help with this test?

Reply:
Started taking a look but for me it also fails in eager mode. How are you running it? (I added `tf.config.run_functions_eagerly(True)` straight after importing tensorflow)

Reply:
Regardless, you'll want to `.assign()` to `._eps` rather than `=` as it's a variable. And just about makes the test pass.